rust/integration-tests/src/sdk_tests/sdk_test_helpers.rs

```diff
@@ -0,0 +1,148 @@
+        },
+        transaction::v1::Transaction,
+    };
+    use futures::Stream;
+    use std::pin::Pin;
+    use tonic::{Request, Response, Status};
+    use diesel::{pg::PgConnection, Connection, RunQueryDsl};
+    use sdk_processor::processors::events_processor::EventsProcessorConfig;
+    use testcontainers::{
+        core::{IntoContainerPort, WaitFor},
+        runners::AsyncRunner,
+        ContainerAsync, GenericImage, ImageExt,
+    };
+    use sdk_processor::config::processor_config::ProcessorConfig;
+    use aptos_indexer_processor_sdk::aptos_indexer_transaction_stream::TransactionStreamConfig;
+    use aptos_indexer_processor_sdk::common_steps::TransactionStreamStep;
+    use sdk_processor::config::db_config::DbConfig;
+    use sdk_processor::config::db_config::PostgresConfig;
+    use sdk_processor::config::indexer_processor_config::IndexerProcessorConfig;
+    use sdk_processor::processors::events_processor::EventsProcessor;
+    use ahash::AHashMap;
+    use anyhow::Context;
+    use url::Url;
+    use crate::models::queryable_models::Event;
+    use crate::utils::mock_grpc::MockGrpcServer;
+    use sdk_processor::schema::events::dsl::*;
+    use diesel::{
+
+        query_dsl::methods::{FilterDsl, ThenOrderDsl},
+        ExpressionMethods
+    };
+
+    type ResponseStream = Pin<Box<dyn Stream<Item = Result<TransactionsResponse, Status>> + Send>>;
+
+    /// Helper function to set up and start Postgres container.
+    async fn setup_postgres() -> (String, ContainerAsync<GenericImage>) {
+        let postgres_container = GenericImage::new("postgres", "14")
+            .with_exposed_port(5432.tcp())
+            .with_wait_for(WaitFor::message_on_stderr(
+                "database system is ready to accept connections",
+            ))
+            .with_env_var("POSTGRES_DB", "postgres")
+            .with_env_var("POSTGRES_USER", "postgres")
+            .with_env_var("POSTGRES_PASSWORD", "postgres")
+            .start()
+            .await
+            .expect("Postgres started");
+
+        let host = postgres_container.get_host().await.unwrap();
+        let port = postgres_container
+            .get_host_port_ipv4(5432)
+            .await
+            .unwrap();
+        let db_url = format!("postgres://postgres:postgres@{host}:{port}/postgres");
+
+        (db_url, postgres_container)
+    }
+
+    /// Helper function to set up and run the mock GRPC server.
+    async fn setup_mock_grpc(transactions: Vec<TransactionsResponse>, chain_id: u64) {
+        let mock_grpc_server = MockGrpcServer {
+            transactions,
+            chain_id,
+        };
+
+        // Start the Mock GRPC server
+        tokio::spawn(async move {
+            println!("Starting Mock GRPC server");
+            mock_grpc_server.run().await;
+        });
+    }
+
+    /// Helper function to initialize the Events Processor.
+    async fn init_events_processor(db_url: String) -> EventsProcessor {
```

Comment:
we may want to explore some generic solutions.

Reply:
There's a mapping between processor -> config and I wonder if we can reuse it here to know which config each processor uses to instantiate the default of that config.

Reply:
if you are thinking of that for SDK migration, I think we can reuse that mapping for us. 
 but i don't think that we can support that for the custom processors.